lib/df-detection.c

```diff
@@ -50,6 +50,14 @@ void add_address(const void *addr, size_t len, unsigned long caller)
 	}
 }
 
+void add_randomization(const void *addr, size_t len)
```

Comment:
As Dmitry pointed out, we shouldn't be doing this, because `addr` is a userspace address.
Because we are calling add_address() after the data has been copied from the userspace, it's already late to change it (the changes would be visible to the kernel only after the third fetch).

I think you need to pass the kernel memory pointer to add_address as well, and add_randomization should be changing the kernel buffer.